### ISSUE TITLE
Fennec doesn't show up in the "Open with ..." menu

### ISSUE THREAD

**Issue by geokon-gh** [mozilla-mobile/focus-android#3202]
## Steps to reproduce
Have Firefox Klar and Firefox Fennec both installed from FDroid.
Set Klar to be the default Browser

### Expected behavior
In the menu bar I expect a "Open in Firefox Fennec"

### Actual behavior
I only get a "Open in ..." menu listing. If I select that then I get an option to either download Firefox or to open using the system Browser (that comes installed with MIUI)

### Extra Note:
The weird thing is that if I set Firefox Fennec to be the default browser then everything works - I get the expected menu item in Klar. But running Fennec as default sorta defeats the point of Klar

### Device information
Xiaomi Note 5 with MIUI
Latest version from FDroid

**Comment by bbinto** [comment]
@brampitoyo to confirm what the expected behaviour should be (intentional?)

**Comment by brampitoyo** [comment]
@geokon-gh This is a bug.

If both Firefox and Firefox Focus are installed, then when you go into the in-app browser’s menu, a menu item called “Open in Firefox” should appear above “Open in…”

If only Firefox Focus is installed, then the in-app browser’s menu will contain the menu item “Open in…”

**Comment by geokon-gh** [comment]
As I explained, I have Firefox Fennec installed - and it doesn't show up when Klar is set as default

**Comment by ekager** [comment]
@brampitoyo I don't think this was ever the case before. If we want to add this as a menu item we can but I have some concerns. It should replace "Open in..." when the user has multiple installed? Doesn't that limit the user's choice? 

If Fennec would be the default we show an "Open in Firefox" option currently, but if there are multiple third party browsers we show "Open in...".

**Comment by brampitoyo** [comment]
@ekager Good point!

What if we set a policy like this:

1. If there’s a default browser set, and it’s other than Focus, then show: “Open in [browserName]” above “Open in…”

2. If Focus is set as a default browser, just show “Open in…”
    * I guess we can try and prioritise Firefox, then put it above “Open in…”, but is that playing favourites?

**Comment by geokon-gh** [comment]
That's all fine @brampitoyo , but I think you guys have lost the point... 

Whether it shows as "Open in Fennec" or in the "Open in ..." menu thing is .. bikesheding. The bug is that Fennec is both not showing up as its own button nor in the "Open in..." menu when Focus/Klar is set as default. There is no way to open the webpage in Fennec at all (but the Xiaomi browser does show up) Maybe I should submit an issue to Fennec? 

To anyone else having this problem - The work around I've found is to hit "Share" then I can open it in Fennec through that menu (it's a bit weird)

**Comment by ekager** [comment]
CC @bbinto @vesta0 @Sdaswani

**Comment by Sdaswani** [comment]
I would ask for 'Help Wanted' on this @ekager .

**Comment by geokon-gh** [comment]
Holy crap.. I dunno how the workflow is at Mozilla.. but it's kinda crazy that this is still open and a one line change has generated a month of discussion haha

@ekager 
I wish I was an Android developer so I could just make the PR for this.. but looking at how other apps behave, whatever Klar is doing is kinda bizarre/non-standard. Like when you open an email address or a link in basically any other app it will pop up a system/native selection menu with the available choices. If you click a link it shows you the browsers... and since the browsers all register with the system as such, the list is automatically populated correctly. So you don't need to maintain a list of valid IDs or anything nuts like that. But maybe there is a good reason you guys are reinventing a slightly-more-square wheel here.... (to have a consistent look across platforms?)

**Comment by Sdaswani** [comment]
@ekager going back to this question: https://github.com/mozilla-mobile/focus-android/issues/3202#issuecomment-420749384

I don’t want to create the expectation we will support every App Store so let’s wait for a contributor to post a fix. If this is broken for google play builds let me know.

**Comment by vesta0** [comment]
@ekager let's go ahead and add the FDroid ID as long as it's a small fix. Thank you.

**Comment by rpappalax** [comment]
this feature doesn't seem to be working yet.

STR
1. Download and install: Klar and "Fennec F-Droid" using the FDroid app
2. Open Settings > Apps & Notifications > Advanced > Default apps
3. Set Klar as default browser 
4. Open a webpage: cnn.com
5. Click on main (3-dot) menu, select "Open in..."

### Expected behavior
6. Fennec F-Droid should appear as an

### Actual behavior
6. Only Chrome appears as an option

![image](https://user-images.githubusercontent.com/3699494/45894449-360d3200-bd83-11e8-8488-af388a4b60b3.png)

![image](https://user-images.githubusercontent.com/3699494/45894430-2aba0680-bd83-11e8-9a37-6cba5dda4cbe.png)
